File: clojure/bob/bob.clj
(ns bob)
(defn response-for [sentence]
  (if (clojure.string/blank? sentence)
    "Fine. Be that way!"
    (if (= sentence (clojure.string/upper-case sentence))
      "Woah, chill out!"
      (if (= (last sentence) \?)
        "Sure."
        "Whatever."))))

File: haskell/bob/Bob.hs
module Bob
  ( responseFor
  ) where

import Data.Char
import Data.List (isSuffixOf)

responseFor :: String -> String
responseFor text
  | isShouting text = "Woah, chill out!"
  | isQuestion text = "Sure."
  | isSilence  text = "Fine. Be that way!"
  | otherwise       = "Whatever."

isSilence :: String -> Bool
isSilence = all isSpace

isShouting :: String -> Bool
isShouting text = not (any isLower text) && any isAlpha text

isQuestion :: String -> Bool
isQuestion = ("?" `isSuffixOf`)
